### PR TITLE
[SYCL][Test] Make testsuite respect SYCL_LIB_DUMPS_ONLY

### PR DESCRIPTION
**Pull request by ldrumm**
Running the check-sycl-dumps target sets the SYCL_LIB_DUMPS_ONLY param for lit which should only enable the ABI dump tests. However, since 1fde65647ebae9 we've added a non `lit.formats.ShTest()` which doesn't respect config.suffixes, so we need to account for this.

### CODE REVIEW

**Comment by ldrumm** [comment]
An alternative to this is to teach the standalone header format about suffixes:

```diff
diff --git a/sycl/test/format.py b/sycl/test/format.py
index 19d16a263426..b64f41844222 100644
--- a/sycl/test/format.py
+++ b/sycl/test/format.py
@@ -23,6 +23,8 @@ class SYCLHeadersTest(lit.formats.TestFormat):
             for filename in filenames:
                 if not filename.endswith(".hpp"):
                     continue
+                if os.path.splitext(filename)[1] not in litConfig.suffixes:
+                    continue
                 filepath = os.path.join(dirpath, filename)

                 if headers_filter is not None:
```

**Comment by AlexeySachkov** [comment]
> An alternative to this is to teach the standalone header format about suffixes:

This might actually be a better approach, because instead of introducing a custom logic for skipping certain tests, it would mean that tests won't launch themselves thanks to existing/standard/built-in mechanism of filters. The latter is a cleaner design, I think

**Comment by ldrumm** [comment]
On Thu Aug 1, 2024 at 10:46 AM BST, Alexey Sachkov wrote:
> The latter is a cleaner design, I think

Yeah. Let's go with that. I'll redo it.

**Review by AlexeySachkov — APPROVED** [review]
Thanks!

File: sycl/test/format.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 import os
 import re
 
+SUFFIXES = {'.hpp'}
```
Comment (ldrumm) [inline sycl/test/format.py:8]: The linter is complaining about my use of single quotes.

image

Seems a bit aggressive, but I'll update

Comment (AlexeySachkov) [reply sycl/test/format.py]: Oh, python linter is very annoying for sure

**Comment by dm-vodopyanov** [comment]
@ldrumm, please add a tag.

https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#commit-message

**Comment by AlexeySachkov** [comment]
> @ldrumm, please add a tag.
> 
> https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#commit-message

Done :)

**Comment by ldrumm** [comment]
> > @ldrumm, please add a tag.
> > https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#commit-message
> 
> Done :)

Thanks Alexey.
As a sidenote: is the file list in the diffstat not enough? These tags seem to eat up a large proportion of the 50 character title and seem to make for worse descriptions in general

**Comment by dm-vodopyanov** [comment]
> > > @ldrumm, please add a tag.
> > > https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#commit-message
> > 
> > 
> > Done :)
> 
> Thanks Alexey. As a sidenote: is the file list in the diffstat not enough? These tags seem to eat up a large proportion of the 50 character title and seem to make for worse descriptions in general

We use these tags at least for creating release notes. @bader can provide more reasons if any.

**Comment by bader** [comment]
The original idea is these patches are supposed to be upstreamed to llvm/llvm-project repository. In order to do that we are trying to follow LLVM's guidelines and policies. This requirement is specified in https://llvm.org/docs/DeveloperPolicy.html#commit-messages.

**Review by dm-vodopyanov — APPROVED** [review]

**Comment by ldrumm** [comment]
> CUDA failures seem down to device detection unrelated to this patch
> 
> @intel/llvm-gatekeepers please merge

sure thing ;)

File: sycl/test/format.py [hunk 2]
```diff
@@ -21,7 +23,8 @@ def getTestsInDirectory(self, testSuite, path_in_suite, litConfig, localConfig):
         for dirpath, _, filenames in os.walk(source_path):
             relative_dirpath = dirpath[len(localConfig.sycl_include) + 1 :]
             for filename in filenames:
-                if not filename.endswith(".hpp"):
+                suffix = os.path.splitext(filename)[1]
+                if suffix not in SUFFIXES or suffix not in litConfig.suffixes:
```
Comment (AlexeySachkov) [inline sycl/test/format.py:27]: @ldrumm, I've just realized that this patch had completely disable the testing suite :)

We don't have `.hpp` registered as a suffix and therefore we always skip all those tests. This is more of a FYI, I'm working on a revamp/expansion of the suite in scope of #16012, so I will fix it as part of that effort

**Pull request merged**